api/database/migrations/2023_08_22_140400_move_category_from_skill_families_to_skill_table.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+use App\Models\Skill;
+use Database\Helpers\ApiEnums;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('skills', function (Blueprint $table) {
+            // Create category column in Skill table, temporarily set category to technical
+            $table->string('category')
+                ->nullable(false)
+                ->default(ApiEnums::SKILL_CATEGORY_TECHNICAL);
+        });
+
+        // Iterate through all skills and update category using data from skill families
+        $skills = Skill::all();
+        foreach ($skills as $skill) {
+            $category = $skill
+                ->families()
+                ->firstWhere('category', '=', ApiEnums::SKILL_CATEGORY_TECHNICAL)
+                ->category;
```

Comment:
Fails to run for me

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/55bb3a76-f1ce-48ba-82f3-cf615a552910)

Adding the null coalesce should take care of it

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/6ab232ed-9a3b-47ac-9d02-e413fba27dbe)

Reply:
Nice catch! Fixed by [28feb55](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7685/commits/28feb55d935f22fdfb9becf254f8edb7eb9c75cd)